bindings/java/src/main/native/context.cc

```diff
@@ -155,27 +152,35 @@ Java_org_openucx_jucx_ucp_UcpContext_memoryMapNative(JNIEnv *env, jobject ctx,
         params.flags = env->GetLongField(jucx_mmap_params, field);;
     }
 
+    if (params.field_mask & UCP_MEM_MAP_PARAM_FIELD_PROT) {
+        field = env->GetFieldID(jucx_mmap_class, "prot", "J");
+        params.prot = env->GetLongField(jucx_mmap_params, field);;
+    }
+
+    if (params.field_mask & UCP_MEM_MAP_PARAM_FIELD_MEMORY_TYPE) {
+        field = env->GetFieldID(jucx_mmap_class, "memType", "I");
+        params.memory_type =
+            static_cast<ucs_memory_type_t>(env->GetIntField(jucx_mmap_params, field));
+    }
+
     ucs_status_t status =  ucp_mem_map((ucp_context_h)ucp_context_ptr, &params, &memh);
     if (status != UCS_OK) {
         JNU_ThrowExceptionByStatus(env, status);
     }
 
-    // Construct UcpMemory class
-    jclass jucx_mem_cls = env->FindClass("org/openucx/jucx/ucp/UcpMemory");
-    jmethodID constructor = env->GetMethodID(jucx_mem_cls, "<init>", "(J)V");
-    jobject jucx_mem = env->NewObject(jucx_mem_cls, constructor, (native_ptr)memh);
+    ucp_mem_attr_t attr = {0};
 
-    // Set UcpContext pointer
-    field = env->GetFieldID(jucx_mem_cls, "context", "Lorg/openucx/jucx/ucp/UcpContext;");
-    env->SetObjectField(jucx_mem, field, ctx);
+    attr.field_mask = UCP_MEM_ATTR_FIELD_ADDRESS | UCP_MEM_ATTR_FIELD_LENGTH |
+                      UCP_MEM_ATTR_FIELD_MEM_TYPE;
```

Comment:
so this we can also detect the type of registered memory?

Reply:
Yes, query all mmap attributes through public API `ucp_mem_query` rather than internal `ucp_mem_h->attr`